packages/react-components/react-datepicker-compat/src/components/DatePicker/useDatePicker.tsx

```diff
@@ -66,27 +66,22 @@ function usePopupVisibility(props: DatePickerProps) {
 }
 
 function useSelectedDate({ formatDate, onSelectDate, value }: DatePickerProps) {
-  const [selectedDate, setSelectedDateState] = useControllableState({
-    initialState: undefined,
+  const [selectedDate, setSelectedDateState] = useControllableState<Date | null | undefined>({
+    initialState: null,
     state: value,
   });
-  const [formattedDate, setFormattedDate] = React.useState(() => (value && formatDate ? formatDate(value) : ''));
-
-  const setSelectedDate = (newDate: Date | undefined) => {
-    if (
-      (selectedDate === undefined && newDate !== undefined) ||
-      (selectedDate !== undefined && newDate === undefined) ||
-      (newDate && selectedDate && (newDate > selectedDate || newDate < selectedDate))
-    ) {
-      onSelectDate?.(newDate);
-    }
+  const [formattedDate, setFormattedDate] = React.useState(() =>
+    value !== null && value && formatDate ? formatDate(value) : '',
+  );
 
+  const setSelectedDate = (newDate: Date | null | undefined) => {
+    onSelectDate?.(newDate);
     setSelectedDateState(newDate);
-    setFormattedDate(newDate && formatDate ? formatDate(newDate) : '');
+    setFormattedDate(newDate !== null && newDate && formatDate ? formatDate(newDate) : '');
```

Comment:
I think checking for `newDate` itself rules out `null`

```suggestion
    setFormattedDate(newDate && formatDate ? formatDate(newDate) : '');
```

Reply:
This is one of those JS being weird things, while you're correct, JS treats null as something. This is why we need the check for null first, otherwise it will return null 😅

![image](https://github.com/microsoft/fluentui/assets/5953191/b46e6746-3f16-4a1d-8f38-814cd8112669)